src/eterna/ui/Toolbar.ts

```diff
@@ -782,6 +788,31 @@ export default class Toolbar extends ContainerObject {
             this.addObject(this.submitButton, this.lowerToolbarLayout);
         }
 
+        // kkk add 3d validation button
+        if (this._type === ToolbarType.PUZZLEMAKER) {
+            this.lowerToolbarLayout.addHSpacer(SPACE_WIDE);
+            this.validate3DButton = new ToolbarButton()
+                .up(Bitmaps.ImgFileOpen)
+                .over(Bitmaps.ImgFileOpenHover)
+                .down(Bitmaps.ImgFileOpen)
```

Comment:
Two notes on this icon:
- It doesn't appear to have a separate hover vs normal graphic with a different color background like the current ones do
- The icon gets the point across that you're picking something from a folder, but nothing about it being a 3D file. The icon should be clearly conveying "upload 3D file".

If you need a new graphic for this to be created, please coordinate with Sharif.